packages/fx-core/src/plugins/resource/cicd/index.ts

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import {
+  err,
+  UserError,
+  SystemError,
+  v2,
+  TokenProvider,
+  FxError,
+  Result,
+  Inputs,
+  Json,
+  Func,
+  ok,
+  QTreeNode,
+  ProjectSettings,
+  SingleSelectQuestion,
+  Platform,
+} from "@microsoft/teamsfx-api";
+
+import { FxResult, FxCICDPluginResultFactory as ResultFactory } from "./result";
+import { CICDImpl } from "./plugin";
+import { ErrorType, InternalError, NoProjectOpenedError, PluginError } from "./errors";
+import { LifecycleFuncNames } from "./constants";
+import { Service } from "typedi";
+import { ResourcePluginsV2 } from "../../solution/fx-solution/ResourcePluginContainer";
+import { ResourcePlugin, Context, DeepReadonly } from "@microsoft/teamsfx-api/build/v2";
+import {
+  githubOption,
+  azdoOption,
+  jenkinsOption,
+  ciOption,
+  cdOption,
+  provisionOption,
+  publishOption,
+  questionNames,
+} from "./questions";
+import { Logger } from "./logger";
+import { environmentManager } from "../../../core/environment";
+import { telemetryHelper } from "./utils/telemetry-helper";
+
+@Service(ResourcePluginsV2.CICDPlugin)
+export class CICDPluginV2 implements ResourcePlugin {
+  name = "fx-resource-cicd";
+  displayName = "CICD";
```

Comment:
I see we already have a constant file packages/fx-core/src/plugins/resource/cicd/constants.ts, can we leverage the value of it instead of define it again?

Reply:
Agree.